server/chat-commands/info.ts

```diff
@@ -2610,13 +2610,15 @@ export const commands: ChatCommands = {
 		}
 
 		this.checkBroadcast(true, '!code');
-
-		const code = Chat.getReadmoreCodeBlock(target);
 		this.runBroadcast(true);
+
 		if (this.broadcasting) {
-			return `/raw <div class="infobox">${code}</div>`;
+			const filteredCode = Chat.filter(this, target, user, room, connection, this.pmTarget);
+			if (!filteredCode) return;
+			target = filteredCode;
+			return `/raw <div class="infobox">${Chat.getReadmoreCodeBlock(target)}</div>`;
```

Comment:
```suggestion
			target = this.filter(target, this.pmTarget);
			if (!target) return this.errorReply(`Invalid code.`);
			return `/raw <div class="infobox">${Chat.getReadmoreCodeBlock(target)}</div>`;
```

Reply:
Less sure on the error, but let's not silently fail.

Reply:
```ts
filter(message: string, targetUser: User | null = null) {
	if (!this.room) return null;
	return Chat.filter(this, message, this.user, this.room, this.connection, targetUser);
}
```
This will always return null in PMs.